R/use_cassette.R

```diff
@@ -175,17 +179,45 @@ use_cassette <- function(
     re_record_interval = re_record_interval,
     clean_outdated_http_interactions = clean_outdated_http_interactions
   )
-  if (is.null(cassette)) {
-    force(...)
-    return(NULL)
-  }
-  on.exit(cassette$eject())
-  cassette$call_block(...)
-  return(cassette)
+
+  # force all arguments
+  list(...)
+
+  invisible(cassette)
 }
 
-check_empty_cassette <- function(cas) {
-  if (!any(nzchar(readLines(cas$file())))) {
-    warning(empty_cassette_message, call. = FALSE)
+#' @rdname use_cassette
+#' @export
+#' @param frame For expert use only: the frame on the call stack that defines
```

Comment:
Can we have an example for how to use frame or maybe point to docs for https://withr.r-lib.org/reference/defer.html ?